tools/autograd/templates/VariableType.cpp

```diff
@@ -124,7 +126,50 @@ struct VariableTypeRegistry {
   }
 };
 
+struct VariableHooks : public at::VariableHooksInterface {
+  VariableHooks(at::VariableHooksArgs) {}
+  void registerVariableTypeFor(at::Context*, at::Backend, at::ScalarType) const override;
+  at::Type& getVariableType(const at::Type&) const override;
+};
+
+// Sigh, the registry doesn't support namespaces :(
+using at::RegistererVariableHooksRegistry;
+using at::VariableHooksRegistry;
+
+// WARNING: YOU MUST DO THE NEXT TWO STATIC INITIALIZERS IN THIS ORDER.
```

Comment:
nit: I'm not a huge fan of comments like this because the expected time for it to be out of date is ~2 days because it's incredibly over specified (it's good for a commit message though!).  Something like "VariableHooks has a dependency on globalContext, which has a dependency on ..." is IMO more appropriate.